jax/_src/numpy/lax_numpy.py

```diff
@@ -3399,14 +3399,15 @@ def argwhere(a, *, size=None, fill_value=None):
 
 
 @util._wraps(np.argmax, skip_params=['out'])
-def argmax(a, axis: Optional[int] = None, out=None, keepdims=None):
-  return _argmax(a, None if axis is None else operator.index(axis), keepdims=bool(keepdims))
-
-@partial(jit, static_argnames=('axis', 'keepdims'), inline=True)
-def _argmax(a, axis: Optional[int] = None, out=None, keepdims=False):
+def argmax(a: ArrayLike, axis: Optional[int] = None, out=None, keepdims=None) -> Array:
   util._check_arraylike("argmax", a)
   if out is not None:
     raise NotImplementedError("The 'out' argument to jnp.argmax is not supported.")
+  return _argmax(asarray(a), None if axis is None else operator.index(axis),
+                 keepdims=bool(keepdims))
+
+@partial(jit, static_argnames=('axis', 'keepdims'), inline=True)
+def _argmax(a: Array, axis: Optional[int] = None, keepdims: bool = False) -> Array:
```

Comment:
Maybe we could drop the default arguments from the inner functions, WDYT?

Reply:
Better would be to drop the inner functions entirely... they're mostly left-over from a time when the functions returned by `jit` didn't support keyword arguments.

Reply:
They do have a benefit, for `partial` doesn't work well with static type-checkers.

Reply:
Yeah, I'm of two minds on that. It's nice to write code that's compatible with static type checkers, but if we're forced to write code in a confusing/inefficient way to appease them, I'd lean towards not jumping through those hoops.

Reply:
We may end up just making a `pyi` file for `jax.numpy` due to issues like this.

Reply:
> We may end up just making a pyi file for jax.numpy due to issues like this.

That's actually great news. And all the overloads will have a place to go.